pkg/core/xds/resource.go

```diff
@@ -5,7 +5,7 @@ import (
 
 	envoy_sd "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
 	envoy_types "github.com/envoyproxy/go-control-plane/pkg/cache/types"
-	protov1 "github.com/golang/protobuf/proto"
+	"google.golang.org/protobuf/runtime/protoimpl"
```

Comment:
> WARNING: This package should only ever be imported by generated messages. The compatibility agreement covers nothing except for functionality needed to keep existing generated messages operational. Breakages that occur due to unauthorized usages of this package are not the author's responsibility.

https://pkg.go.dev/google.golang.org/protobuf@v1.35.2/runtime/protoimpl
I need to use one public API

Reply:
I noticed that in the previous Mike's PR https://github.com/kumahq/kuma/pull/4834 regenerated the objects(proto message v2).

So we don't need to use this internal package voluntarily and we can also remove `protov1.MessageV1` function to convert a `v1/v2` message to a `v2` message.